src/systems/towers.js

```diff
@@ -0,0 +1,33 @@
+import { System, Not } from 'ecsy';
+import { Tile, Tower } from '../components';
+
+export default class TowerSystem extends System {
+  execute(delta) {
+    this.queries.newTowers.results.forEach(t => {           
+        let tower = t.getComponent(Tower);
+        let matchingTileEntity;
+        this.queries.emptyTiles.results.forEach(et => {                              
+            let tile = et.getComponent(Tile);
+            if (tile.x == tower.x && tile.y == tower.y) {                   
+                matchingTileEntity = et;
+            }
+        });
```

Comment:
You could do this instead:

```js
let matchingTileEntity = this.queries.emptyTiles.results.find(et => {
  let { x, y } = et.getComponent(Tile);
  return x === tower.x && y === tower.y;
});
```

`find` will also exit as soon as there's a match instead of looping over e'rything.